marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -72,9 +62,30 @@ def _bq_table_name(self, bq_table):
     def extract_on_complete(self, task_instance) -> [StepMetadata]:
         log.debug(f"extract_on_complete({task_instance})")
 
-        source = self._source()
-        bigquery_job_id = self._get_bigquery_job_id(task_instance)
+        try:
+            source = self._source()
+            bigquery_job_id = self._get_bigquery_job_id(task_instance)
+        except Exception as e:
+            log.error(f"Cannot retrieve job details from BigQuery.Client. {e}",
+                      exc_info=True)
+            try:
+                sql_meta = SqlParser.parse(self.operator.sql)
+                log.debug(f"bigquery sql parsed and obtained meta: {sql_meta}")
 
+                return [StepMetadata(
+                    name=get_job_name(task=self.operator),
+                    context={
+                        'sql': self.operator.sql,
+                        'bigquery.sql.parsed.inputs':
+                            json.dumps(sql_meta.in_tables),
+                        'bigquery.sql.parsed.outputs':
+                            json.dumps(sql_meta.out_tables)
+                    }
+                )]
+            except Exception as e:
+                log.error(f"Cannot parse sql query. {e}",
```

Comment:
I think in this case we want to add the error to the context, skip parsing and keep going like we do for `bigquery.extractor.error` bellow

Reply:
added it to the context and i have it emitting a new step so it can be properly resolved